crates/matrix-sdk/tests/integration/room/joined.rs

```diff
@@ -560,3 +560,27 @@ async fn fetch_members_deduplication() {
         results.iter().filter(|r| r.as_ref().unwrap().as_ref().unwrap().is_some()).count();
     assert_eq!(response_count, 1);
 }
+
+#[async_test]
+async fn set_name() {
+    let (client, server) = synced_client().await;
+
+    mock_sync(&server, &*test_json::SYNC, None).await;
+    let sync_settings = SyncSettings::new();
+    client.sync_once(sync_settings).await.unwrap();
+
+    let room = client.get_joined_room(&test_json::DEFAULT_SYNC_ROOM_ID).unwrap();
+    let name = "The room name";
+
+    Mock::given(method("PUT"))
+        .and(path_regex(r"^/_matrix/client/r0/rooms/.*/state/m.room.name/$"))
```

Comment:
Don't we want to [expect] the path to be called? Otherwise we're not testing much here.

[expect]: https://docs.rs/wiremock/latest/wiremock/struct.Mock.html#method.expect

Reply:
We wouldn't have a response if the function didn't call the right endpoint with the right data.

Reply:
Hmm, but the test doesn't assert that we got a response. Sure, the function currently returns the response, but we can mutate the function to not return a response and this test will continue on working.

We need to either check that we received what we expect or make sure the endpoint actually gets called.